bundles/org.openhab.binding.nuki/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -101,5 +101,21 @@
 			</options>
 		</state>
 	</channel-type>
+	<channel-type id="smartlockDoorState">
+		<item-type>Number</item-type>
```

Comment:
Channels with options should be of item-type String.

Reply:
The new channel only has numbers as value. I copied the configuration from the smartlockState which is exactly the same.

Reply:
Hmm that's not how it should be. I think in practice it's not really a problem at this moment. See here in the documentation about the information it needs to be a String: https://www.openhab.org/docs/developer/bindings/thing-xml.html#state-description

Reply:
I understand. I will test locally and give feedback asap.

Reply:
I have searched a bit, but cannot find an example in the repository where an item-type string is used while the option values are integers. As far as I can see from the state-description, an item-type string is only useful if the values are also strings.  

Also the conversion from an integer to a state is quite strange and it seems it has to be done like this: 
`updateState(channelDoorState.getUID(), (State) new StringType(Integer.toString(NukiBindingConstants.DOORSENSOR_STATES_UNKNOWN)));`

Reply:
I just tested a bit more. When keeping the org.eclipse.smarthome.core.types.State library, I could not cast the integer values to string states. When using the org.openhab.core.types.State library the already existing code throws a lot of errors about type conversion. 

Summary: I would like to keep the type as Number for this rather small addon. Maybe with the OpenHab 3 a conversion makes more sense.

Reply:
To get to the bottom of this I've asked about what it should be and it turns out Number is actual valid. So this is perfectly fine and the documentation is wrong on this. So I'm going to merge it as is.

FYI. the State in org.openhab is from the openHAB 1 compatibility layer and should not be used. In openHAB 3 the namespace of core will be changed from eclipse to openhab. That will mean if you look at openHAB 3 source code the State in org.eclipse is gone, and the org.openhab is than the version that was originally in org.eclipse. The openHAB 1 compatibility layer will be removed in openHAB 3.